assignments/array-methods.js

```diff
@@ -56,22 +56,38 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+
+runners.forEach(function(currentvalue){
+    fullName.push(currentvalue.first_name.concat( " " +currentvalue.last_name));
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+runners.map(function(currentvalue){
+    allCaps.push(currentvalue.first_name.toUpperCase());
+})
+
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+largeShirts.push(
+  runners.filter(function(big){
+    return big.shirt_size === 'L';
+  })
+)
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+let ticketPriceTotal = runners.reduce(function(accumulator, currentValue){
+    return accumulator + currentValue.donation;
+  },0)
+  console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
```

Comment:
Here are a few more examples:

// Problem 1
/** LIST OF RUNNERS IMPORTANT INFORMATION */
    let importantInfo = [];
    runners.forEach(runner => importantInfo.push(`First Name: ${runner.first_name} Last Name: ${runner.last_name} Email: ${runner.email} Company: ${runner.company_name}`));
    console.log(importantInfo);

// Problem 2
/** List of companies who attended the event and sort it*/
    let companyList = runners.map(runner => runner.company_name);
    console.log(companyList.sort());

// Problem 3
/** list of all shirts ordered */
    let shirtOrders = runners.map(runner => runner.shirt_size);

    let shirt2XL = runners.filter(person => person.shirt_size === "2XL");
    let shirtXL = runners.filter(person => person.shirt_size === "XL");
    let shirtL = runners.filter(person => person.shirt_size === "L");
    let shirtM = runners.filter(person => person.shirt_size === "M");
    let shirtS = runners.filter(person => person.shirt_size === "S");
    console.log(`
        2XL shirt order: ${shirt2XL.length}
        XL shirt order: ${shirtXL.length}
        L shirt order: ${shirtL.length}
        M shirt order: ${shirtM.length}
        S shirt order: ${shirtS.length}
    `);

// Problem 3 
    console.log(shirtOrders);